Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM ubuntu:16.04 as builder
+LABEL author="Abdelrahman Hosny <abdelrahman.hosny@hotmail.com>"
+
+RUN apt-get update && apt-get install -y build-essential \
+    clang \
+    bison \
+    flex \
+    libreadline-dev \
+    gawk \
+    tcl-dev \
+    libffi-dev \
+    git \
+    graphviz \
+    xdot \
+    pkg-config \
+    python3
+
+COPY . /
+RUN make && \
+    make install
+
+
+FROM ubuntu:16.04
+RUN apt-get update && apt-get install -y clang \
```

Comment:
Do we need clang, bison, flex, libreadline-dev, tcl-dev, libffi-dev, git, pkg-config at runtime?

Reply:
I only kept `libreadline-dev` and `tcl-dev` and removed the rest. They give the following errors when removed:
`error while loading shared libraries`